### PR TITLE
#1511 #1507

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#1511 Encoding list in the import dialog may appear empty
#1507 VS crash: ObjectDisposedException for MarshalingWindowFrame

### CODE REVIEW

File: src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs [hunk 1]
```diff
@@ -162,6 +164,12 @@ public ImportDataWindow(string filePath, string name)
         }
 
         private async Task PreviewAsync() {
+            if(_encodingsInitTask == null) {
```
Comment (int19h) [inline src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs]: It looks like it can never be null?

**Comment by int19h** [comment]
:+1:

File: src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs]: It is null when InitializeComponent creates combobox and it fires selection change

File: src/Package/Impl/PackageManager/VsRPackageManagerVisualComponent.cs [hunk 2]
```diff
@@ -27,13 +30,9 @@ internal class VsRPackageManagerVisualComponentContainerFactory : ToolWindowPane
...
-            var workflow = _workflowProvider.GetOrCreate();
-            if (workflow.ActiveWindow == null) {
-                workflow
-                    .GetOrCreateVisualComponent(_componentContainerFactory)
-                    .ContinueOnRanToCompletion(w => w.Container.Show(false));
-            }
-
+            var o = new object();
+            var shell = VsAppShell.Current.GetGlobalService<IVsUIShell>(typeof(SVsUIShell));
+            shell.PostExecCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdShowReplWindow, 0, ref o);
```
Comment (AlexanderSher) [inline src/Package/Impl/PackageManager/VsRPackageManagerVisualComponent.cs]:
```{csharp}
var workflow = _workflowProvider.GetOrCreate();
if (workflow.ActiveWindow == null) {
    VsAppShell.Current.DispatchOnMainThreadAsync(() => workflow.GetOrCreateVisualComponent(_componentContainerFactory));
}
```

File: src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs (hunk 1 above)
Comment (AlexanderSher) [inline src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs]: yep, it is safe to just call `await _encodingsInitTask;`

**Pull request merged**